renovate.json

```diff
@@ -1,5 +1,15 @@
 {
   "extends": [
     "config:base"
+  ],
+  "packageRules": [
+    {
+      "matchPackageNames": ["org.codehaus.groovy:groovy-dateutil", "org.codehaus.groovy:groovy-all"],
+      "allowedVersions": "<3"
+    },
+    {
+      "matchPackageNames": ["org.spockframework:spock-core"],
+      "allowedVersions": "/^[0-9]+\\.[0-9]+(\\.[0-9]+)?-groovy-2\\.5$/"
```

Comment:
spock provides per version suffix groovy versions, it would be easier if these were different artifacts as renovate tries to update to groovy 3.x with this 😢 

hopefully this works, I tested via regex tester and looked at the docs 
https://docs.renovatebot.com/configuration-options/#allowedversions